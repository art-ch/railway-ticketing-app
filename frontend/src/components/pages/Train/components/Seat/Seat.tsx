'use client';

import React from 'react';

import { Seat as SeatType } from 'railway-ticketing-app-sdk';
import Link from 'next/link';

export type Props = { trainId: string; seat: SeatType };

export const Seat = ({ trainId, seat }: Props) => {
  const { seatNumber, isBooked } = seat;

  const commonStyles =
    'border border-slate-300 rounded-sm p-1 text-center text-5xl cursor-pointer  text-white';

  const freeSeat = 'bg-emerald-800';
  const bookedSeat = 'bg-amber-500';

  const bookingStateStypes = isBooked ? bookedSeat : freeSeat;

  return (
    <Link
      className={`${commonStyles} ${bookingStateStypes}`}
      href={`/${trainId}/${seatNumber}`}
    >
      {seatNumber}
    </Link>
  );
};
