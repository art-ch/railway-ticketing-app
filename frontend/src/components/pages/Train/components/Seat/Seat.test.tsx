import { render } from '@testing-library/react';
import { Seat } from './Seat';

describe('Seat component', () => {
  it('should render correctly when seat is not booked', () => {
    const { container } = render(
      <Seat trainId="1" seat={{ seatNumber: 1, isBooked: false }} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly when seat is booked', () => {
    const { container } = render(
      <Seat
        trainId="1"
        seat={{ seatNumber: 1, isBooked: true, bookingId: '1' }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
