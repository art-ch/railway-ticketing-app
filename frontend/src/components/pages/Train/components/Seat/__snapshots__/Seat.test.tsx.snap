// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seat component should render correctly when seat is booked 1`] = `
<div>
  <a
    class="border border-slate-300 rounded-sm p-1 text-center text-5xl cursor-pointer text-white bg-amber-500"
    href="/1/1"
  >
    1
  </a>
</div>
`;

exports[`Seat component should render correctly when seat is not booked 1`] = `
<div>
  <a
    class="border border-slate-300 rounded-sm p-1 text-center text-5xl cursor-pointer text-white bg-emerald-800"
    href="/1/1"
  >
    1
  </a>
</div>
`;
